File: labs/solidworks/swex/pmpage/controls/selection-box/SelectionBox.CustomFilter.cs
public class SelectionBoxCustomSelectionFilterDataModel
{
    public class DataGroup
    {
        [SelectionBox(typeof(PlanarFaceFilter), swSelectType_e.swSelFACES)] //setting the standard filter to faces and custom filter to only filter planar faces
        public IFace2 PlanarFace { get; set; }
    }

    public class PlanarFaceFilter : SelectionCustomFilter<IFace2>
    {
        protected override bool Filter(IPropertyManagerPageControlEx selBox, IFace2 selection, swSelectType_e selType, ref string itemText)
        {
            itemText = "Planar Face";
            return selection.IGetSurface().IsPlane(); //validating the selection and only allowing planar face
        }
    }
}

File: labs/solidworks/swex/pmpage/controls/selection-box/SelectionBox.CustomFilter.vb
Public Class SelectionBoxCustomSelectionFilterDataModel

    Public Class DataGroup
        <SelectionBox(GetType(PlanarFaceFilter), swSelectType_e.swSelFACES)>
        Public Property PlanarFace As IFace2
    End Class

    Public Class PlanarFaceFilter
        Inherits SelectionCustomFilter(Of IFace2)

        Protected Overrides Function Filter(ByVal selBox As IPropertyManagerPageControlEx, ByVal selection As IFace2, ByVal selType As swSelectType_e, ByRef itemText As String) As Boolean
            itemText = "Planar Face"
            Return selection.IGetSurface().IsPlane()
        End Function
    End Class

End Class